core/server/master/src/main/java/alluxio/master/file/cmdmanager/FileSystemCommandManager.java

```diff
@@ -0,0 +1,371 @@
+
+/**
+ * The Filesystem Command manager which controls distributed job operations.
+ */
+public final class FileSystemCommandManager implements DelegatingJournaled {
+  private static final Logger LOG = LoggerFactory.getLogger(FileSystemCommandManager.class);
+  private final CmdEntry mCmdEntry = new CmdEntry();
+  private final Map<Long, CommandStatus> mStatusMap = Maps.newHashMap();
+  private final Scheduler mScheduler = new Scheduler(null);
+
+  /**
+   * Constructor.
+   */
+  public FileSystemCommandManager() {}
+
+  /**
+   * Validate the command information.
+   * @param commandInfo command information
+   * @return boolean value on whether the command is validated for scheduling or not
+   */
+  public boolean validate(CommandInfo commandInfo) {
+    CmdOptions options = commandInfo.getCmdOptions();
+    String path = commandInfo.getPath();
+    ValidationStatus status = mCmdEntry.getValidationStatus(path, options);
+
+    if (status.equals(ValidationStatus.New)) {
+      return true;
+    }
+
+    if (status.equals(ValidationStatus.Update_Options)) {
+      CommandInfo info = mCmdEntry.mCommandPathToInfo.get(path);
+      long commandId = info.getId();
+      /*Just update command status map with new options.*/
+      CommandStatus commandStatus = mStatusMap.get(commandId);
+      commandStatus.updateOptions(options);
+    }
+    return false;
+  }
+
+  /**
+   * Schedule a command to run.
+   * @param commandInfo command meta information
+   * @throws ResourceExhaustedException throw ResourceExhaustedException
+   * @throws InterruptedException throw InterruptedException
+   */
+  public void schedule(CommandInfo commandInfo)
+          throws ResourceExhaustedException, InterruptedException {
+    long commandId = commandInfo.getId();
+    CommandStatus commandStatus = new CommandStatus(commandInfo.getId(),
+            commandInfo.getPath(), commandInfo.getCmdOptions());
+    mScheduler.schedule(commandStatus);
+    mStatusMap.put(commandId, commandStatus);
+  }
+
+  /**
+   * Get status for a command.
+   * @param id command id
+   * @return running status
+   */
+  public Status getStatus(Long id) {
+    return null;
+  }
+
+  /**
+   * Stop a command for running.
+   * @param id command id
+   */
+  public void stop(Long id) {
+  }
+
+  /**
+   * Close the command manager.
+   * @throws IOException IOException
+   */
+  public void close() throws IOException {
+  }
+
+  @Override
+  public Journaled getDelegate() {
+    return mCmdEntry;
+  }
+
+  static class Scheduler {
+    private static final int CAPACITY = 100;
+    private static final long TIMEOUT = 100;
+    private final ExecutorService mExecutorService = Executors.newSingleThreadExecutor();
+    private final BlockingQueue<CommandStatus> mCommandQueue = new LinkedBlockingQueue<>(CAPACITY);
+    private int mCurrentSize = 0;
+    private final Client mClient;
+    private long mTimeoutCounts;
+    private long mRTCounts;
+
+    public Scheduler(Client client) {
```

Comment:
Please read through the bugout scheme documentation, specifically on constructors.